esmvalcore/cmor/_fixes/cmip6/miroc6.py

```diff
@@ -1,11 +1,41 @@
 """Fixes for MIROC6 model."""
-from ..common import ClFixHybridPressureCoord
+import numpy as np
 
+from ..common import ClFixHybridPressureCoord
+from ..fix import Fix
 
 Cl = ClFixHybridPressureCoord
 
-
 Cli = ClFixHybridPressureCoord
 
-
 Clw = ClFixHybridPressureCoord
+
+
+class Tos(Fix):
+    """Fixes for tos."""
+
+    def fix_metadata(self, cubes):
+        """
+        Fix latitude_bounds and longitude_bounds data type and round to 4 d.p.
+
+        Parameters
+        ----------
+        cubes : iris.cube.CubeList
+            Input cubes.
+
+        Returns
+        -------
+        iris.cube.CubeList
+        """
+        for cube in cubes:
+            latitude = cube.coord('latitude')
+            if latitude.bounds is not None:
+                latitude.bounds = latitude.bounds.astype(np.float32)
```

Comment:
Is this large data? The convention is to use `float64` (https://github.com/ESMValGroup/ESMValCore/blob/2e7e3848b0a7ab1bd5485605eb0d151d3dbb8bab/esmvalcore/cmor/tables/cmip6/Tables/CMIP6_coordinate.json#L837) for coordinates.

Reply:
Yes, we looked into that. The problem is that scenario data for this dataset comes with coordinates in float32 whereas the historical one is in float64, so the concatenation of the two experiments failed. 

We tried converting the scenario to float64 and the concatenation was impossible due to rounding errors. The only way we found to be able to work with both experiments was by setting less precision in the coordinates.

Reply:
Yes, I can see that you implemented the rounding on line 34 and I think that's fine. But why are you choosing to convert the datatype fo float32 instead of float64?

Reply:
If it is converted to float64 the coordinates that are in 32 get some nonsense values at the last decimal places and it is not possible to round it afterwards so that they match with float64 values. Not sure why, though

Reply:
It looks like you're only rounding the bounds and not the points, at least in the code I can see here. How about rounding the points to the same precision?

Reply:
Yes, we are aware, but there is no rounding needed for the points if we convert to float32.

Nevertheless, if we convert to float64, even if we round we get differences in the historical and ssp coordinates. Here is a debug, note the max and min values of the differences. (the top difference is between the bounds and the lower is the points)
![Screenshot from 2023-05-30 16-38-45](https://github.com/ESMValGroup/ESMValCore/assets/66776374/2d3eb916-467c-4633-88ac-cb34c772b458)

In this case the fix is as follows (note that I added the points rounding just in case):
```python
class Tos(Fix):
    """Fixes for ** all vars."""

    def fix_metadata(self, cubes):
        """
        ***Fix latitude_bounds and longitude_bounds data type and round to 4 d.p.

        Parameters
        ----------
        cubes : iris.cube.CubeList
            Input cubes.

        Returns
        -------
        iris.cube.CubeList

        """
        for cube in cubes:
            latitude = cube.coord('latitude')
            if latitude.bounds is None:
                latitude.guess_bounds()
            latitude.bounds = latitude.bounds.astype(np.float64)
            latitude.bounds = np.round(latitude.bounds, 4)
            latitude.points = latitude.points.astype(np.float64)
            latitude.points = np.round(latitude.points, 4)
            longitude = cube.coord('longitude')
            if longitude.bounds is None:
                longitude.guess_bounds()
            longitude.bounds = longitude.bounds.astype(np.float64)
            longitude.bounds = np.round(longitude.bounds, 4)
            longitude.points = longitude.points.astype(np.float64)
            longitude.points = np.round(longitude.points, 4)
        return cubes
```

Reply:
If just converting to `float32` already does the trick, how about doing something like:
```
latitude.points = latitude.points.astype(np.float32).astype(np.float64)
```
?

Reply:
I agree, this is a very sexy move.

Nevertheless, it invents the data from the 6th decimal in the points and bounds. it will probably not influence the results, but do you think we should allow this?

example input lat points:  [23.0211550, 26.3700972]
example outout lat points: [23.021154403686523, 26.370098114013672]

its in the meters scale so it shouldn't be a big problem

Reply:
It's no worse then inventing zeros and what else can you do if the modellers don't provide the data? The more precise alternative would be add the correct values from a file that has them.